x-pack/plugins/ingest_manager/server/services/epm/packages/remove.ts

```diff
@@ -20,7 +20,10 @@ export async function removeInstallation(options: {
   // TODO:  the epm api should change to /name/version so we don't need to do this
   const [pkgName] = pkgkey.split('-');
   const installation = await getInstallation({ savedObjectsClient, pkgName });
-  const installedObjects = installation?.installed || [];
+  if (!installation) throw new Error('integration does not exist');
+  if (installation.removable === false)
+    throw new Error(`The ${pkgName} integration is installed by default and cannot be removed`);
```

Comment:
Could we throw BadRequest error so we have a 400 it's a user error?

Reply:
If someone tries to do this through the API, what response code does he get currently?

Reply:
@ruflin 500

Reply:
@jfsiii Any thoughts on which error code we should return?

Overall, I would not block the PR on which error code we should return. Which API error codes we return where we should probably have a general look.

Reply:
@nchaulet Because the new NP framework interface does not have native support for converting Boom exceptions into HTTP responses I am thinking we need some kind of helper for this.  Otherwise we are having to check for each type of error? https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/routes/agent/handlers.ts#L57  . Perhaps we can always through Boom errors and always return custom errors that pass along Boom results?

Reply:
Yes it need some custom code in the handler I did it like this  for some fleet APIs https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/routes/agent/handlers.ts#L122